ethcore/builtin/src/lib.rs

```diff
@@ -211,89 +227,128 @@ impl Builtin {
 
 	/// Whether the builtin is activated at the given block number.
 	pub fn is_active(&self, at: u64) -> bool {
-		at >= self.activate_at
+		self.pricer.keys().rev().any(|&other_at| at >= other_at)
 	}
 }
 
 impl From<ethjson::spec::Builtin> for Builtin {
 	fn from(b: ethjson::spec::Builtin) -> Self {
-		let pricer: Box<dyn Pricer> = match b.pricing {
-			ethjson::spec::Pricing::Blake2F { gas_per_round } => {
-				Box::new(gas_per_round)
+		// TODO(niklasad1): change this to `try_from` and propogate the error accordingly
+		let native = EthereumBuiltin::from_str(&b.name).unwrap();
```

Comment:
I get the following error message for this:

```bash
Thread 'main' panicked at 'invalid builtin name: foo', ethcore/builtin/src/lib.rs:282
```

which I thought is equivalent to a manual panic? 

Otherwise, I have to change to have `Box<dyn Trait` instead of an enum.

Reply:
Ah, ok, so `unwrap` for `Result` panic with the proper message, but I would rather have an explicit panic instead of unwrap, since unwrap usually indicates that it cannot panic in practice, otherwise it's a bug.

Reply:
added an explicit panic